x/orderbook/bucket.go

```diff
@@ -26,6 +31,7 @@ type OrderBookBucket struct {
 func NewOrderBookBucket() *OrderBookBucket {
 	b := morm.NewModelBucket("orderbook", &OrderBook{},
 		morm.WithIndex("market", marketIDindexer, false),
+		morm.WithIndex("marketWithTickers", marketIDTickersIndexer, false),
```

Comment:
In theory we could remove the market index, as a query with `("/orderbook/market", 00000005)` and `("/orderbook/market?prefix", 00000005)` would return the same data. This would make the API a bit more complex, but save some space on the server.

You can leave with the two indexes (second one unique), but add a comment that the first is somewhat redundant, and to remove it if we can still provide a usable client-side API without it.

Reply:
Can you send me couple examples where querying happens? Probably somewhere [iov-core](https://github.com/iov-one/iov-core) but I need some guidance :smiley: I need to examine and get my head around the concept.

Reply:
Raw ABCI query path is [parsed here](https://github.com/iov-one/weave/blob/03670052c46e70d16dd2d4e16cf90ea03a8409ad/app/store.go#L221-L223). We [look up the registered handler](https://github.com/iov-one/weave/blob/03670052c46e70d16dd2d4e16cf90ea03a8409ad/app/store.go#L223-L224) for the path, then [dispatch to the proper handler](https://github.com/iov-one/weave/blob/03670052c46e70d16dd2d4e16cf90ea03a8409ad/app/store.go#L250-L251) along with the modifier (eg. "prefix")

This is handled by the orm.Bucket, which [provides a Query() method](https://github.com/iov-one/weave/blob/03670052c46e70d16dd2d4e16cf90ea03a8409ad/orm/bucket.go#L116-L137)

Reply:
Re: iov-core

We can query one object, eg. [getAccount](https://github.com/iov-one/iov-core/blob/master/packages/iov-bns/src/bnsconnection.ts#L345-L347)

We can query by prefix scan, eg. [getAllTokens](https://github.com/iov-one/iov-core/blob/master/packages/iov-bns/src/bnsconnection.ts#L329-L331)

We can also query by secondary index, eg. [to look up atomic swaps](https://github.com/iov-one/iov-core/blob/master/packages/iov-bns/src/bnsconnection.ts#L414-L423)

There is no example in iov-core of using prefix mod on secondary indexes, but there is no reason to believe it would be difficult, after it was properly documented and explained

Reply:
Will examine all and add this links to bestpractices :+1: